### PR TITLE
Merging geocoded records

### PR DESCRIPTION
**Pull request by adrian-lara**
#### Any background context you want to provide?
Merging geocoded records did not trigger the merging of geocoding results properly.

The solution to this involves using the complete geocoding results entirely from one record. For example, we're preventing the use of the Latitude of one record with the Longitude of the other.

See the linked issue for more details.

#### What's this PR do?
Allows for geocoding results between records to be merged when the two records are merged.

From the user perspective, adding merge protection to one of these geocoding result columns adds merge protection to the other geocoding result columns.

#### How should this be manually tested?
Merge geocoded records between changes of the merge protection for the geocoding result columns - Longitude, Latitude, and Geocoding Confidence.

#### What are the relevant tickets?
#2018

### CODE REVIEW

File: seed/lib/merging/tests/test_merging.py [hunk 1]
```diff
@@ -151,6 +156,167 @@ def test_merge_state_favor_existing(self):
         self.assertEqual(result.address_line_2, 'new')
         self.assertEqual(result.extra_data['field_1'], 'orig_value')
 
+    def test_merge_geocoding_results_no_merge_protection(self):
```
Comment (nllong) [inline seed/lib/merging/tests/test_merging.py]: woah! TDD.. love it! 😜

File: seed/lib/merging/merging.py [hunk 2]
```diff
@@ -84,6 +88,50 @@ def get_state_attrs(state_list):
...
+        existing_results_empty = existing_results_empty and can_attrs[geocoding_col][state1] is None
+        new_results_empty = new_results_empty and can_attrs[geocoding_col][state2] is None
+
+        geocoding_favor_new = geocoding_favor_new and priorities.get(geocoding_col, 'Favor New') == 'Favor New'
+
+        # Since these are handled here, remove them from canonical attributes
+        del can_attrs[geocoding_col]
+
+    # Multiple elif's here is necessary since empty checks should be first, followed by merge protection settings
+    if new_results_empty:
```
Comment (nllong) [inline seed/lib/merging/merging.py]: seems like possible performance issues here... we'll cross that bridge when we get there.

**Comment by adrian-lara** [comment]
> Maybe this isn't related, but I can't finish testing due to the following.
> 
> I've created a geocoding profile but the site isn't allowing me to map asserting that I need a Custom ID 1... which I have.
> 
> image
> 
> Maybe this is fixed elsewhere?

I wasn't able to reproduce this. I'm thinking there's a lower chance that this is related, but would you be able to share the file you were using during import? 

Also, it looks like you're mapping both "Custom ID 1" and "Address Line 1", so it's even more surprising that you're getting that error.

**Comment by RDmitchell** [comment]
@adrian-lara / @nllong  -- I have seen this problem. See issue #2054

**Comment by nllong** [comment]
Going to accept this one as the issue associated with it has been addressed. The other issue #2054 has been captured and in the queue to be fixed.

**Pull request merged**